src/main/java/core/basesyntax/serviceimpl/ReportServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportService;
+import java.util.Map;
+
+public class ReportServiceImpl implements ReportService {
+    private static final String SEPARATOR = ",";
+    private static final String HEADER = "fruit,quantity";
+
+    @Override
+    public String createReport() {
+        StringBuilder stringBuilder = new StringBuilder(HEADER);
+        for (Map.Entry<String, Integer> entry : Storage.storage.entrySet()) {
+            stringBuilder.append(System.lineSeparator())
+                    .append(entry.getKey())
+                    .append(SEPARATOR)
+                    .append(entry.getValue());
```

Comment:
The `Storage.storage` field is directly accessed here. It's better to use a service or DAO layer for such interactions to follow the encapsulation principle.